code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/docker/OpendDockerComposeFile.xtend

```diff
@@ -0,0 +1,242 @@
+
+/**
+ * This class uses the singleton design pattern handling the generation of the docker compose file.
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+class OpendDockerComposeFile {
+    static OpendDockerComposeFile instance
+    String dockerComposePath
+    Map<String, String> dockerComposeServiceParts
+    List<String> dockerComposeNetworkParts
+
+
+    new() {
+        dockerComposeServiceParts = new HashMap
+        dockerComposeNetworkParts = new LinkedList
+    }
+
+    /*
+     * Get or create instance of the [OpendDockerComposeFile] class.
+     */
+    static def OpendDockerComposeFile getInstance() {
+        if (OpendDockerComposeFile.instance === null)
+            OpendDockerComposeFile.instance = new OpendDockerComposeFile
+
+        return OpendDockerComposeFile.instance
+    }
+
+    /*
+     * This method initializes the [OpendDockerComposeFile] class. Therefor the taget folder for the
+     * docker compose file is set. Additionally, an existing docker compose file is loaded from the
+     * file system into the [DockerComposeFile] date structure.
+     */
+    def init(String targetFolder) {
+        dockerComposePath = targetFolder + File.separator + "docker-compose.yml"
+    }
+
+    /*
+     * Load a existing docker compose file form the file system. If there is no docker compose file
+     * present a default network configuration is created, because LEMMA's operation modeling
+     * language does in its actual version not support the handling of network configurations.
+     */
+    def openExistingDockerComposeFile() {
+        if (new File(dockerComposePath).exists && new File(dockerComposePath).length > 0) {
+            loadDockerComposePartsFromFile
+        } else {
+            createDefaultNetwork
+        }
+    }
+
+    /*
+     * Add or replace a docker compose part to the docker compose file based on the name of the
+     * operation node.The configuration of the docker compose part is created by an intermediate
+     * deployment aspect.
+     */
+    def addOrReplaceDockerComposePart(String name, String dockerComposeServicePart) {
+        this.addOrReplaceToMap(name,
+            Util::adjustIndentations('  ', dockerComposeServicePart) + '\n'
+        )
+    }
+
+    /*
+     * Add or replace a docker compose part to the docker compose file based on the name of the
+     * operation node. The configuration of the docker compose part is created by the docker compose
+     * code generation module itself.
+     *
+     * Note: the docker compose part is done separately for intermediate container and intermediate
+     * infrastructure nodes.
+     */
+    def addOrReplaceDockerComposePart(IntermediateOperationNode node) {
+        if (IntermediateInfrastructureNode.isInstance(node)) {
+            addOrReplaceToMap(node.name,
+                getServiceFromInfrastructureNode(node as IntermediateInfrastructureNode) + "\n"
+            )
+        } else {
```

Comment:
What if we introduce a third type for operation nodes in the future? Then, the subsequent call would fail (`ClassCastException`). You could consider introducing an `else if (node instanceof IntermediateContainer)` here, BUT Xtend has a nice shorthand `switch`-based type guard syntax (https://www.eclipse.org/xtend/documentation/203_xtend_expressions.html#switch-expression) for such issues:
```
switch(node) {
IntermediateInfrastructureNode : ...
IntermediateContainer : ...
default : throw Illegal ArgumentException
}
```